web/src/main/java/io/fabric8/launcher/web/endpoints/OpenShiftEndpoint.java

```diff
@@ -70,7 +72,20 @@
             Set<OpenShiftCluster> subscribedClusters = clusterRegistry.getSubscribedClusters(securityContext.getUserPrincipal());
             for (OpenShiftCluster cluster : subscribedClusters) {
                 futures.add(identityProvider.getIdentityAsync(authorization, cluster.getId())
-                                    .thenApply(identity -> new ClusterVerified(cluster, identity.isPresent())));
+                                    .thenApply(identity -> {
+                                        if (!identity.isPresent()) {
+                                            return new ClusterVerified(cluster, false);
+                                        }
+                                        final ImmutableParameters.Builder builder = ImmutableParameters.builder().cluster(cluster).identity(identity.get());
+                                        final OpenShiftService service = openShiftServiceFactory.create(builder.build());
```

Comment:
We should fail this line if the authorization token is invalid. Perhaps the constructor should call getLoggedUser() to test that. 

Better move it to the try/catch block.

Reply:
I'm not sure what your asking, do you want the construction of the service to already call openshift so that the "user" of the service knows it's a valid token? What is wrong with the way it is now?

Reply:
My point is that the `OpenShiftService` is created with an invalid identity, the create must fail

Reply:
@gastaldi I am not sure you want a request to be made when building a service (neither an exception to be thrown). I think that the call itself should fail, as it actually does..

Reply:
@ia3andy what's the point of having a Service with an invalid identity set where you can't perform any operations? Or maybe I am missing something

Reply:
This is what I mean: https://paste.fedoraproject.org/paste/3kcdI9BVmvCDBTvtqkFzjg

Reply:
@gastaldi having a factory doing a request (or anything beside creating the object it self) is not very nice (IMO).

Reply:
@ia3andy IMHO the factory must be able to create a valid object. As I mentioned earlier, If the identity is invalid, there is nothing that can be done with the created `OpenShiftService`, so it's better to fail early.

Reply:
well it's how the openshiftService works currently and IMO it's nice that it's not doing http requests when you only want to create a service. So I feel changing this behavior should not be part of this PR, this is just to validate if the connected clusters are really connected with a valid token.

Reply:
Ok, I'll change it anyway